### PR TITLE
create docker chs development web

### PR DESCRIPTION
**Pull request by HatiDev**
### JIRA link
https://companieshouse.atlassian.net/browse/LP-142

### Change description

create docker chs development web

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes look good on mobile
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by markpit — APPROVED** [review]

File: .dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+*
+!src
+!views
```
Comment (makhtar-ch) [inline .dockerfile:3]: If we're including the src directory and all its contents, do we need !views? Plus, I'm wondering if this should be a .dockerignore file, I'll get some clarity on this and get back to you.

Comment (markpit) [reply .dockerfile:3]: yes i think you are right, the contents looks like a .dockerignore file, so we'll need one of those and delete the .dockerfile , I don't think it's needed as we have Dockerfile

Comment (HatiDev) [reply .dockerfile]: I have deleted the .dockerfile file and created the .dockerignore file.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: .dockerfile (hunk 1 above)
Comment (HatiDev) [reply .dockerfile]: @makhtar-ch  did you get any additional clarity?

Comment (makhtar-ch) [reply .dockerfile]: yeah Mark pretty much covered it above.

**Review by markpit — APPROVED** [review]

**Review by makhtar-ch — COMMENTED** [review]
@markpit do we need an executable start script?

**Comment by HatiDev** [comment]
> @markpit do we need an executable start script?

@makhtar-ch  do you mean as part of this subtask?

**Comment by markpit** [comment]
> @markpit do we need an executable start script?

To get the app running in docker then yes, as the Dockerfile is making a call to CMD ["./docker_start.sh"]

**Comment by makhtar-ch** [comment]
> > @markpit do we need an executable start script?
> 
> @makhtar-ch do you mean as part of this subtask?

I was just asking in general, could be part of this task or separate.

**Comment by HatiDev** [comment]
> > > @markpit do we need an executable start script?
> > 
> > 
> > @makhtar-ch do you mean as part of this subtask?
> 
> I was just asking in general, could be part of this task or separate.

I have added script for docker

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: docker_start.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+PORT=3000
+
+export NODE_PORT=${PORT}
+exec node /opt/server.js -- ${PORT}
```
Comment (lduranteau) [inline docker_start.sh:5]: I have a doubt about this line.
Is it not rather:
node opt/dist/bin/www.js

Comment (markpit) [reply docker_start.sh:5]: i think the COPy command in Dockerfile will copy the contents of dist, so you won't need dist in the path

Comment (HatiDev) [reply docker_start.sh]: edited line

[the remark this answers is not in the document]
Comment (HatiDev) [reply docker_start.sh]: line changed to use www.js instead of server.js

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+*
+!src
+!views
```
Comment (markpit) [inline .dockerignore:3]: the !views can be removed, we don't have that folder anymore

Comment (HatiDev) [reply .dockerignore]: views removed

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by markpit — APPROVED** [review]

**Review by lduranteau — APPROVED** [review]

**Pull request merged**